src/set_relation/isl_str_manipulation.cc

```diff
@@ -0,0 +1,291 @@
+/*!
+ * \file isl_str_manipulation.h
+ *
+ * \Implementations of isl string manipulator functions
+ *
+ * \date Started: 2016-03-11
+ *
+ * \authors Michelle Strout, Mahdi Soltan Mohammadi
+ *
+ * Copyright (c) 2016, University of Arizona <br>
+ * All rights reserved. 
+ * See ../../COPYING for details. <br>
+ */
+
+#include "isl_str_manipulation.h"
+
+namespace iegenlib{
+
+
+//****** ISL tuple correction functions
+
+/*! This function extracts tuple variables from a Tuple Declaration string.
+**  Tuple Declaration can be either of Relation or Set form
+**  Input:  "[i1,col_tv3_] -> [i3,i4]" or
+**          "[i1,col_tv3_,i3,i4]"  
+**  Output: (pointer to)
+**          {"i1", "col_tv3_", "i3", "i4"}
+*/
+std::string* tupVarsExtract(std::string tupDecl, int inArity, int outArity){
+
+    int st=0, end=0, idxSt = 0;
+    int nparts=1;
+    int arity = inArity + outArity;
+    std::string* tupVars = new std::string[arity];
```

Comment:
Let's not create a string on the heap.  Just return the type std::string.  This will be consistent with the rest of the library when it comes to string and avoiding the need to do cleanup of strings.